jerry-core/jrt/jrt.h

```diff
@@ -31,6 +31,21 @@
 /*
  * Attributes
  */
+#ifdef _MSC_VER
+#define __noreturn __declspec(noreturn)
+#define __attr_noinline___ __declspec(noinline)
+#define __attr_return_value_should_be_checked___
+#define __attr_hot___
+#ifndef __attr_always_inline___
```

Comment:
Is it possible that some variant of MSVC (pre)defines `__attr_always_inline___`, `__attr_const___`, or `__attr_pure___ `? If not, the `#ifndef`s are unnecessary.

Reply:
Question, why the orignal version have #ifdef ?
That's just a copy of the orignal version.
```
#define __noreturn __attribute__((noreturn))
#define __attr_noinline___ __attribute__((noinline))
#define __attr_return_value_should_be_checked___ __attribute__((warn_unused_result))
#define __attr_hot___ __attribute__((hot))
#ifndef __attr_always_inline___
#define __attr_always_inline___ __attribute__((always_inline))
#endif /* !__attr_always_inline___ */
#ifndef __attr_const___
#define __attr_const___ __attribute__((const))
#endif /* !__attr_const___ */
#ifndef __attr_pure___
#define __attr_pure___ __attribute__((pure))
#endif /* !__attr_pure___ */
```

Reply:
Good question. This code dates back to the very beginning of the project (https://github.com/jerryscript-project/jerryscript/commit/afe242e7a3d96ac1353fecc1bc43eea5649bb5c3). For some 3.5 years now, nobody has dared to touch this. Most probably, the original `#ifndefs` can be removed as well. May be a candidate for a follow-up refactoring patch.